contracts/EscrowedNori.sol

```diff
@@ -0,0 +1,725 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.13;
+
+import "@openzeppelin/contracts-upgradeable/utils/math/MathUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC777/IERC777RecipientUpgradeable.sol";
+import "./ERC777PresetPausablePermissioned.sol";
+import "./BridgedPolygonNORI.sol";
+import "./Removal.sol";
+import {RemovalUtils} from "./RemovalUtils.sol";
+
+/*
+Open Questions:
+- do we need to set up an escrow schedule with the full amount associated with the agreement?
+  or is it ok that the amount of the escrow schedule actually changes over time either with the addition
+  of new NORI or with the revocation of some?
```

Comment:
For optimal defi compatibility (e.g., w/ lending platforms etc) I think it's important that `balanceOf(address)` returns the full amount they will eventually receive (which probably means that the schedule needs to be created in full, but funded as it goes). 

Unwrapping to NORI is probably the only thing that needs to use a a different `balanceOf` function that is invoked before unwrapping

Reply:
as discussed on call w Scott, we're going to leave as-is for now, where balanceOf only returns the total number of funded tokens that are backed by actual NORI for now, and look into layering solutions that might help suppliers leverage the value of their likely future tokens.